src/shared/modes.ts

```diff
@@ -12,6 +12,16 @@ export type ModeConfig = {
 	groups: readonly ToolGroup[] // Now uses groups instead of tools array
 }
 
+// Mode-specific prompts only
+export type PromptComponent = {
+	roleDefinition?: string
+	customInstructions?: string
+}
+
+export type CustomPrompts = {
+	[key: string]: PromptComponent | undefined | string
```

Comment:
I don't love this. Can we keep the support prompts as a separate type from the mode-specific prompts?